plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/jdt/ui/wizards/NewInterfaceWizardPage.java

```diff
@@ -0,0 +1,202 @@
+		new LabeledText(this, "Name:").setText(name);
+		return this;
+	}
+
+	/**
+	 * Sets a given package name.
+	 * 
+	 * @param packageName Package name
+	 */
+	public NewInterfaceWizardPage setPackage(String packageName) {
+		new LabeledText(this, "Package:").setText(packageName);
+		return this;
+	}
+
+	/**
+	 * Sets a given source folder.
+	 * 
+	 * @param sourceFolder Source folder
+	 */
+	public NewInterfaceWizardPage setSourceFolder(String sourceFolder) {
+		new LabeledText(this, "Source folder:").setText(sourceFolder);
+		return this;
+	}
+
+	/**
+	 * Returns a package name.
+	 * 
+	 * @return package name
+	 */
+	public String getPackage() {
+		return new LabeledText(this, "Package:").getText();
+	}
+
+	/**
+	 * Returns a interface name.
+	 * 
+	 * @return Interface name
+	 */
+	public String getName() {
+		return new LabeledText(this, "Name:").getText();
+	}
+
+	/**
+	 * Returns a source folder.
+	 * 
+	 * @return Source folder
+	 */
+	public String getSourceFolder() {
+		return new LabeledText(this, "Source folder:").getText();
+	}
+
+	/**
+	 * Check/Uncheck Enclosing type.
+	 *
+	 * @param toggle enclosing type checkbox
+	 */
+	public NewInterfaceWizardPage toggleEnclosingTypeCheckBox(boolean toggle) {
+		new CheckBox("Enclosing type:").toggle(toggle);
+		return this;
+	}
+
+	/**
+	 * Returns Enclosing type.
+	 * 
+	 * @return Enclosing type
+	 */
+	public String getEnclosingType() {
+		return new DefaultText(2).getText();
+	}
+
+	/**
+	 * Sets a given Enclosing type.
+	 * 
+	 * @param enclosingTypeName Enclosing type
+	 */
+	public NewInterfaceWizardPage setEnclosingType(String enclosingTypeName) {
+		new DefaultText(2).setText(enclosingTypeName);
+		return this;
+	}
+
+	/**
+	 * Add extended interface
+	 * 
+	 * @param interfaceName String with name of interface to add
+	 */
+	public void addExtendedInterface(String interfaceName) {
+		new PushButton(3).click();
+
+		String activeShellTitleText = new DefaultShell().getText();
+		if (!activeShellTitleText.equals("Extended Interfaces Selection")) {
+			throw new RedDeerException(
+					"Current opened shell should be labeled with title 'Extended Interfaces Selection'.");
+		}
+
+		new DefaultText(0).setText(interfaceName);
+
+		switch (new DefaultTable(0).getItems().size()) {
+		case 0:
+			new PushButton("Cancel").click();
+			throw new RedDeerException("No item was found for given interface name '" + interfaceName + "'.");
+		case 1:
+			new PushButton("OK").click();
+			break;
+		default:
+			new PushButton("Cancel").click();
+			throw new RedDeerException("More than 1 item was found for given interface name '" + interfaceName + "'.");
+		}
+	}
+
+	/**
+	 * Remove extended interface
+	 * 
+	 * @param interfaceName String with name of interface to remove
+	 */
+	public void removeExtendedInterface(String interfaceName) {
+		DefaultTable table = new DefaultTable(0);
+
+		try {
+			table.getItem(interfaceName).select();
+		} catch (CoreLayerException e) {
```

Comment:
I am thinking of removing the try-catch block... it would be better maybe to just keep that getItem().select there as CoreLayerException going from not found item is saying something very similar what you got there in RDException...

Reply:
Try-catch blocks removed.